collection-scripts/mce-gather.sh

```diff
@@ -211,25 +211,29 @@ gather_all_hostedclusters() {
 }
 
 gather_service_and_event_logs_for_failed_agents() {
-  GATHER_DIR="$1"
-  find "${GATHER_DIR}" -path '*/agentclusterinstalls/*.yaml' -type f | while read -r file; do
-    dir=$(dirname "$file")
-    base=$(basename -s .yaml "$file")
+  gather_dir=${1}
+  agent_cluster_install_list=$(oc get agentclusterinstalls.extensions.hive.openshift.io --all-namespaces -o json)
+  total_installs=$(echo "${agent_cluster_install_list}" | jq '.items | length' )
 
-    # yq eval '.status.conditions[] | select(.type == "Failed") | .status'  must-gather/namespaces/local-cluster/extensions.hive.openshift.io/agentclusterinstalls/local-cluster-cluster-install.yaml
-    failed=$(yq eval '.status.conditions[] | select(.type == "Failed") | .status' "$file")
+  for idx in $(seq 0 $((total_installs - 1))); do
+    install_json=$(echo "${agent_cluster_install_list}" | jq '.items['${idx}']')
+    install_name=$(echo "${install_json}" | jq -r '.metadata.name')
+    install_namespace=$(echo "${install_json}" | jq -r '.metadata.namespace')
+    dir=${gather_dir}/namespaces/${install_namespace}/extensions.hive.openshift.io/agentclusterinstalls
+
+    failed=$(echo "${install_json}" | jq -r '.status.conditions[] | select(.type == "Failed") | .status')
     if [ "$failed" = "False" ]; then
-      echo "skipping logs and events for non-failed cluster $file"
+      log "skipping logs and events for non-failed cluster ${install_name}"
```

Comment:
The `log` function isn't available in the MCE must-gather (yet).
```suggestion
      echo "skipping logs and events for non-failed cluster ${install_name}"
```